homeassistant/components/tplink/entity.py

```diff
@@ -28,12 +28,11 @@
 from homeassistant.helpers.update_coordinator import CoordinatorEntity
 
 from . import legacy_device_id
-from .const import DOMAIN, PRIMARY_STATE_ID
```

Comment:
Any time we have an import of platform into the base entity module this is a hint that we have a design issue where the abstraction of the sensor and light platform is leaking into the base entity.
```
from homeassistant.components.light import LightEntity
from homeassistant.components.sensor import SensorEntity
```

Reply:
Done in https://github.com/home-assistant/core/pull/120025